cvprac/cvp_api.py

```diff
@@ -1308,6 +1310,51 @@ def add_note_to_configlet(self, key, note):
         return self.clnt.post('/configlet/addNoteToConfiglet.do',
                               data=data, timeout=self.request_timeout)
 
+    def sanitize_warnings(self, data):
+        ''' Sanitize the warnings returned after validation.
+            
+            In some cases where the configlets has both errors
+            and warnings, CVP may split any warnings that have 
+            `,` across multiple strings.
+            This method concats the strings back into one string
+            per warning, and correct the warningCount. 
+
+            Args:
+                data (dict): A dict that contians the result
+                    of the validation operation
+            Returns:
+                response (dict): A dict that contains the result of the
+                    validation operation
+        '''
+        if "warnings" not in data:
+            # nothing to do here, we can return as is
+            return data
+        # Since there may be warnings incorrectly split on
+        # ', ' within the warning text by CVP, we join all the 
+        # warnings together using ', ' into one large string
+        temp_warnings = ", ".join(data['warnings']).strip()
+
+        # To split the large string again we match on the 
+        # 'at line XXX' that should indicate the end of the warning.
+        # We capture as well the remaining \\n or whitespace and include
+        # the extra ', ' added in the previous step in the matching criteria.
+        # The extra ', ' is not included in the strings of the new list
+        temp_warnings = split(
+            r'(.*?at line \d+.*?),\s+',
+            temp_warnings
+        )
+
+        # The behaviour of re.split will add empty strings
+        # if the regex matches on the begging or ending of the lin.
```

Comment:
Typo in comment? line?

Reply:
Correct, this is a typo. Update this line in the latest commit.